### DISCUSSION TITLE
Questions about relationship concepts

### DISCUSSION THREAD

**Discussion by natoverse** [microsoft/graphrag#792] [Algorithm + Paper]
Moving from https://github.com/microsoft/graphrag/issues/788

I opened the generated GraphML files in Gephi, and unlike the entities, it seems that the relationships are not typed and only have descriptions. My questions are:

1. Do these relationship descriptions play a role in the query process?
2. Can I set typed relationships, as we do with entities, or is GraphRAG designed to work without relationship types?

**Comment by natoverse** [comment]
1. Yes, the relationship descriptions are included in the summarization for local search (configurable top _k_ relationships related to matched input entity).
2. We do not establish any relationship types during indexing, and this is not configurable. You could potentially tune the graph extraction prompt to add some form of relationship type to the description, but it would not be a structured data field like entity type. The "type" displayed in your Gephi screenshot is part of the graphml definition, and just indicates that it is an undirected graph.

**Comment by MiratPH** [comment]
Thank you, it's really helpful.
